metagpt/rag/engines/simple.py

```diff
@@ -0,0 +1,259 @@
+    NodeWithScore,
+    QueryBundle,
+    QueryType,
+    TransformComponent,
+)
+
+from metagpt.rag.factories import (
+    get_index,
+    get_rag_embedding,
+    get_rankers,
+    get_retriever,
+)
+from metagpt.rag.interface import NoEmbedding, RAGObject
+from metagpt.rag.llm import get_rag_llm
+from metagpt.rag.retrievers.base import ModifiableRAGRetriever, PersistableRAGRetriever
+from metagpt.rag.retrievers.hybrid_retriever import SimpleHybridRetriever
+from metagpt.rag.schema import (
+    BaseIndexConfig,
+    BaseRankerConfig,
+    BaseRetrieverConfig,
+    BM25RetrieverConfig,
+    ObjectNode,
+)
+from metagpt.utils.common import import_class
+
+
+class SimpleEngine(RetrieverQueryEngine):
+    """SimpleEngine is designed to be simple and straightforward.
+
+    It is a lightweight and easy-to-use search engine that integrates
+    document reading, embedding, indexing, retrieving, and ranking functionalities
+    into a single, straightforward workflow. It is designed to quickly set up a
+    search engine from a collection of documents.
+    """
+
+    def __init__(
+        self,
+        retriever: BaseRetriever,
+        response_synthesizer: Optional[BaseSynthesizer] = None,
+        node_postprocessors: Optional[list[BaseNodePostprocessor]] = None,
+        callback_manager: Optional[CallbackManager] = None,
+        index: Optional[BaseIndex] = None,
+    ) -> None:
+        super().__init__(
+            retriever=retriever,
+            response_synthesizer=response_synthesizer,
+            node_postprocessors=node_postprocessors,
+            callback_manager=callback_manager,
+        )
+        self.index = index
+
+    @classmethod
+    def from_docs(
+        cls,
+        input_dir: str = None,
+        input_files: list[str] = None,
+        transformations: Optional[list[TransformComponent]] = None,
+        embed_model: BaseEmbedding = None,
+        llm: LLM = None,
+        retriever_configs: list[BaseRetrieverConfig] = None,
+        ranker_configs: list[BaseRankerConfig] = None,
+    ) -> "SimpleEngine":
+        """From docs.
+
+        Must provide either `input_dir` or `input_files`.
+
+        Args:
+            input_dir: Path to the directory.
+            input_files: List of file paths to read (Optional; overrides input_dir, exclude).
+            transformations: Parse documents to nodes. Default [SentenceSplitter].
+            embed_model: Parse nodes to embedding. Must supported by llama index. Default OpenAIEmbedding.
+            llm: Must supported by llama index. Default OpenAI.
+            retriever_configs: Configuration for retrievers. If more than one config, will use SimpleHybridRetriever.
+            ranker_configs: Configuration for rankers.
+        """
+        if not input_dir and not input_files:
+            raise ValueError("Must provide either `input_dir` or `input_files`.")
+
+        documents = SimpleDirectoryReader(input_dir=input_dir, input_files=input_files).load_data()
```

Comment:
is this Reader fixed for this Engine? If so, you can write it as an attribute, otherwise, would a config be better?

Reply:
the extend demands can be proposed if doesn't meet the current usage.